pkg/splunk/enterprise/util_test.go

```diff
@@ -331,3 +335,265 @@ func TestGetSmartstoreRemoteVolumeSecrets(t *testing.T) {
+	}
+
+	// Test-5: Changing the AppSource deployment info should change for all the Apps in the list
+	changeAppSrcDeployInfoStatus(appFramworkConf.AppSources[0].Name, appDeployContext.AppsSrcDeployStatus, enterprisev1.RepoStateDeleted, enterprisev1.DeployStatusPending, enterprisev1.DeployStatusInProgress)
+	err = validateAppSrcDeployInfoByStateAndStatus(appFramworkConf.AppSources[0].Name, appDeployContext.AppsSrcDeployStatus, enterprisev1.RepoStateDeleted, enterprisev1.DeployStatusInProgress)
+	if err != nil {
+		t.Errorf("Invalid AppSrc deployment info detected. Error: %v", err)
+	}
+
+	// Test-6: When an App is deleted on remote store, it should be marked as deleted
+	setStateAndStatusForAppDeployInfoList(appDeployContext.AppsSrcDeployStatus[appFramworkConf.AppSources[0].Name].AppDeploymentInfoList, enterprisev1.RepoStateActive, enterprisev1.DeployStatusPending)
+
+	// delete an object on remote store for the app source
+	tmpS3Response := S3Response
+	tmpS3Response.Objects = append(tmpS3Response.Objects[:0], tmpS3Response.Objects[1:]...)
+	remoteObjListMap[appFramworkConf.AppSources[0].Name] = tmpS3Response
+
+	retStatus, err = handleAppRepoChanges(client, &cr, &appDeployContext, remoteObjListMap, &appFramworkConf)
+	if !retStatus || err != nil {
+		t.Errorf("Could not handle a valid remote listing. Error: %v", err)
+	}
+
+	err = validateAppSrcDeployInfoByStateAndStatus(appFramworkConf.AppSources[0].Name, appDeployContext.AppsSrcDeployStatus, enterprisev1.RepoStateActive, enterprisev1.DeployStatusPending)
+	if err != nil {
+		t.Errorf("Unable to delete/disable an app when the App is deleted from remote store. Error: %v", err)
+	}
+
+	// Test-7: Object hash change on the remote store should cause App state and status as Active and Pending.
+	S3Response.Objects = createRemoteObjectList("e41d8cd98f00", startAppPathAndName, 2322, nil, 10)
+	remoteObjListMap[appFramworkConf.AppSources[0].Name] = S3Response
+
+	setStateAndStatusForAppDeployInfoList(appDeployContext.AppsSrcDeployStatus[appFramworkConf.AppSources[0].Name].AppDeploymentInfoList, enterprisev1.RepoStateDeleted, enterprisev1.DdeployStatusComplete)
+
+	retStatus, err = handleAppRepoChanges(client, &cr, &appDeployContext, remoteObjListMap, &appFramworkConf)
+	if !retStatus || err != nil {
+		t.Errorf("Could not handle a valid remote listing. Error: %v", err)
+	}
+
+	err = validateAppSrcDeployInfoByStateAndStatus(appFramworkConf.AppSources[0].Name, appDeployContext.AppsSrcDeployStatus, enterprisev1.RepoStateActive, enterprisev1.DeployStatusPending)
+	if err != nil {
+		t.Errorf("Unable to detect the change, when the object changed. Error: %v", err)
+	}
+
+	// Test-8:  For an AppSrc, when all the Apps are deleted on remote store and re-introduced, should modify the state to active and pending
+	setStateAndStatusForAppDeployInfoList(appDeployContext.AppsSrcDeployStatus[appFramworkConf.AppSources[0].Name].AppDeploymentInfoList, enterprisev1.RepoStateDeleted, enterprisev1.DdeployStatusComplete)
```

Comment:
A test like this might be helpful
// Test-8.1: For an AppSrc, when an App is deleted on remote store, updated offline and re-introduced, should modify the App state to active and Deployment status to pending

Reply:
I think test-7 is the same as the scenario.